services/core/shared/core/compat/sdk_v5/coreCache.js

```diff
@@ -13,41 +13,27 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
-const { Utils, CacheLRU } = require('lisk-service-framework');
-
-const { isEmptyArray } = Utils.Data;
-const { getAccounts } = require('./coreApi');
+const { CacheLRU } = require('lisk-service-framework');
 
 const config = require('../../../../config');
+const { knex } = require('../../../database');
 
 const cache = CacheLRU();
 const getCacheKey = (key, value) => `account:${key}:${value}`;
 
-const normalizeAccount = account => {
-	account.address = account.address.toString('hex');
-	account.token.balance = Number(account.token.balance);
-	account.sequence.nonce = Number(account.sequence.nonce);
-	account.dpos.sentVotes = account.dpos.sentVotes.map(vote => {
-		vote.delegateAddress = vote.delegateAddress.toString('hex');
-		vote.amount = Number(vote.amount);
-		return vote;
-	});
-
-	return account;
-};
-
 const getCachedAccountBy = async (key, value) => {
+	const accountsDB = await knex('accounts');
 	const cacheKey = getCacheKey(key, value);
 	let account = await cache.get(cacheKey);
 	if (!account) {
-		const result = await getAccounts({ [key]: value });
-		if (!Array.isArray(result.data) || isEmptyArray(result.data)) {
+		const result = await accountsDB.find({ [key]: value });
+		if (result === undefined) {
 			const expireMiliseconds = config.ttl.affectedByNewBlocks;
 			await cache.set(cacheKey, null, expireMiliseconds);
 			return null;
 		}
-		const { address, dpos: { delegate: { username } } } = normalizeAccount(result.data[0]);
-		account = { address, username };
+		const [{ address, username, publicKey }] = result;
```

Comment:
With the current implementation, for scenarios when `accountDB.find()` doesn't return any values, the existing line of code would throw errors. Please pair this change with the one above or another appropriate alternative for them both.
```suggestion
		const { address, username, publicKey } = result;
```